README.md

```diff
@@ -1,2 +1,308 @@
+
+This method sets the fast charge current limit.
+
+The default fast charge current limit is 2048mA.
+#### Parameters ####
+
+| Parameter | Type | Required | Description |
+| --- | --- | --- | --- |
+| *currentLimit* | Integer | Yes | The desired fast charge current limit in milli Amps (0 - 5056mA)
+
+#### Return Value ####
+
+None.
+
+#### Example ####
+
+```squirrel
+// Sets the fast charge current limit to 5056mA
+batteryCharger.setCurrentLimit(5056);
+```
+### setChargeVoltage(*milliVolts*) ###
+
+This method sets the desired battery voltage that the device should charge to.
+
+The default charge voltage is 4352 mV.
+#### Parameters ####
+
+| Parameter | Type | Required | Description |
+| --- | --- | --- | --- |
+| *chargeVoltage* | Integer | Yes | The desired charge voltage in milli Volts (3840 - 4608mV)|
+
+#### Return Value ####
+
+None.
+
+#### Example ####
+
+```squirrel
+// Sets the charge voltage to 4200mV
+batteryCharger.setChargeVoltage(4200);
+```
+
+
+
+### getChargeVoltage() ###
+
+Returns the target battery charge voltage
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+
+local voltage = batteryCharger.getChargeVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+
+### getBatteryVoltage() ###
+
+Returns the current battery voltage based on the internal ADC conversion.
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getBatteryVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getVBUSVoltage() ###
+
+Returns the VBUS voltage based on the ADC conversion, this is the input voltage
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getVBUSVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getSystemVoltage() ###
+
+Returns the system voltage based on the ADC conversion, this the output voltage which can be used to power other chips in your application
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getSystemVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getChargingCurrent() ###
+
+Returns the measured current going to the battery 
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The charging current in milli Amps.
+
+#### Example ####
+
+```squirrel
+local current = batteryCharger.getChargingCurrent();
+server.log("Current: " + current + "mA");
+```
+### getChargingStatus() ###
+
+ Returns the charging status
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — Not Charging = 0,  Pre-charge = 1, Fast Charging = 2, Charge Termination Good = 3
+```squirrel
```

Comment:
I would write this a bit differently. 

#### Return Value ####

Integer - see table below for supported values.

| Charging Status Constant | Value | 
| --- | --- | 
| BQ25895M_CHARGING_STATUS.NOT_CHARGING | 0x00 | 
| BQ25895M_CHARGING_STATUS.PRE_CHARGE | 0x08 |
| BQ25895M_CHARGING_STATUS.FAST_CHARGE | 0x10 |
| BQ25895M_CHARGING_STATUS.CHARGE_TERMINATION_DONE | 0x18 |